dataikuapi/dss/ml.py

```diff
@@ -308,50 +332,679 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+        assert isinstance(stratified, bool)
+        self._raw_settings["stratified"] = stratified
+
+    def set_validation_n_folds(self, n_folds):
+        assert isinstance(n_folds, int)
+        if self._raw_settings["mode"] not in {"KFOLD", "TIME_SERIES_KFOLD"}:
+            warnings.warn("\'nFolds\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["nFolds"] = n_folds
+
+    def set_validation_split_ratio(self, split_ratio):
+        assert isinstance(split_ratio, float)
+        if self._raw_settings["mode"] not in {"SHUFFLE", "TIME_SERIES_SINGLE_SPLIT"}:
+            warnings.warn("\'splitRatio\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["splitRatio"] = split_ratio
+
+    def set_validation_n_shuffle_iter(self, n_shuffle_iter):
+        assert isinstance(n_shuffle_iter, int)
+        if self._raw_settings["mode"] != "SHUFFLE":
+            warnings.warn("\'shuffleIterations\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["shuffleIterations"] = n_shuffle_iter
+
+    def set_execution_timeout(self, timeout):
+        assert isinstance(timeout, int)
+        self._raw_settings["timeout"] = timeout
+
+    def set_execution_n_iter(self, n_iter):
+        assert isinstance(n_iter, int)
+        if self._raw_settings["strategy"] == "GRID":
+            self._raw_settings["nIter"] = n_iter
+        else:
+            self._raw_settings["nIterRandom"] = n_iter
+
+    def set_grid_search_shuffling(self, shuffled, seed=None):
+        assert isinstance(shuffled, bool)
+        if seed is not None:
+            assert isinstance(seed, int)
+            self._raw_settings["seed"] = seed
+        self._raw_settings["randomized"] = shuffled
+        if self._raw_settings["strategy"] != "GRID":
+            warnings.warn("\'randomized\' and \'seed\' parameters will be ignored in {} strategy".format(self._raw_settings["strategy"]))
+
+    def set_search_distribution(self, distributed, n_containers):
+        assert isinstance(distributed, bool)
+        if n_containers is not None:
+            assert isinstance(n_containers, int)
+            self._raw_settings["nContainers"] = n_containers
+        self._raw_settings["distributed"] = distributed
+
+    def set_n_jobs(self, n_jobs):
+        assert isinstance(n_jobs, int) and n_jobs > 0
+        self._raw_settings["nJobs"] = n_jobs
+
+
+class HyperparameterSettings(object):
+
+    def __init__(self, name, algo_settings):
+        self.name = name
+        self._algo_settings = algo_settings
+
+
+class NumericalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        clean_dict = self._algo_settings._get_active_numerical_hyperparam_dict(self.name)
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(clean_dict, indent=4))
```

Comment:
It's debatable but I don't think it's an issue to make the logs _more_ readable, and since they often end up being searched with tools such as less or grep anyways, having more lines should not be a problem.
The idea here is to provide relevant and readable information to the user at a glance, especially when they are designing experiments in notebooks.

Reply:
Hmm, specifically, if we make grep-ability a priority over readability, that would be an argument for remaining on a single line, as `cat my.log | grep some-expr` will by default only display that line and not display the other lines form that object. Whereas if the whole object is on a single line, it will get displayed. But a single line JSOn is harder to read…

Reply:
In this case I'd tend to favor greppability / one-line in string repr (precisely because log processing tools work better with one line), but readability / multi-line in HTML for notebooks (including details `<pre>`).